src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DirectoryHelper.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public static class DirectoryHelper
+    {
+        public static IEnumerable<string> GetFilteredFiles(string workspaceDirectory, string searchPattern, IReadOnlyCollection<string> ignoredNames, ILogger? logger)
+        {
+            IEnumerable<string> files;
+            try
+            {
+                files = Directory.GetFiles(workspaceDirectory, searchPattern, SearchOption.TopDirectoryOnly);
```

Comment:
Seems like EnumerateFiles would be a better choice here. Given a structure like this:

```
foo
foo\bar.txt
foo\AReallyLongPathThatExceeedsTheMaxLength.txt
foo\baz.txt
```

If I understand correctly the current code will throw at calling `GetFiles("foo")` so the calling code won't see `bar.txt` or `baz.txt`, or anything in any subdirectories of foo.

Reply:
I'm fine to switch to EnumerateFiles, and you're right about leaving off things after a Too-Long file, but I'm not trying to make the LongPath situation air-tight. My goal is to avoid the most likely causes of failure on this path and try to handle anything outside of that without crashing VS. As I said in the issue for this, once you've violated the Windows path length requirements you've gone off the rails and our ability to protect you is limited.

Does that pass muster? Also open to better designs here.